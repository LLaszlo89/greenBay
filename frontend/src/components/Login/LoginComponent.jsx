import React from "react";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import Dialog from "@material-ui/core/Dialog";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import { Typography } from "@material-ui/core";
import sendHttpRequest from "../../apiRequest";

const URL = "http://localhost:3000/api/session";

const initialState = {
  name: "",
  password: "",
};

export default function Login() {
  const [open, setOpen] = React.useState(false);
  const [values, setValues] = React.useState(initialState);
  const [error, setError] = React.useState(false);
  const [errorMessage, setErrorMessage] = React.useState("");

  const handleError = () => {
    setError(true);
  };

  const handleLoginToggle = () => {
    setOpen(!open);
  };
  const handleInputChange = (e) => {
    const { name, value } = e.target;
    setValues({
      ...values,
      [name]: value,
    });
  };

  const onSubmit = async () => {
    const user = {
      username: values.name,
      password: values.password,
    };
    const response = await sendHttpRequest("POST", URL, user);
    if (!response.token) {
      handleError();
      setErrorMessage(response)
    } else {
      console.log(response);
    }
    //   REDIRECT TO SHOP
    console.log(response);
  };

  return (
    <div>
      <Typography align={"center"}>
        Welcome to the coolest web shop on the planet earth , to enjoy all the
        benefits please log in
      </Typography>
      <Typography align={"center"}>
        <Button variant="outlined" color="primary" onClick={handleLoginToggle}>
          Login
        </Button>
      </Typography>
      <Dialog open={open} onClose={handleLoginToggle} fullWidth>
        <DialogTitle id="form-dialog-title">
          Please enter your login details:{" "}
        </DialogTitle>
        <Typography align="center" variant="h4" color="error" >{ errorMessage }</Typography>
        <DialogContent>
          <TextField
            placeholder="Enter Your Name"
            label="Name"
            name="name" // Without it will not update state!!!!!!!!!!!
            value={values.name}
            onChange={handleInputChange}
            margin="normal"
            fullWidth
            error={error}
          />
          <br />
          <TextField
            placeholder="Enter Your Password"
            label="Password"
            name="password" //state name so we can use it
            value={values.password}
            onChange={handleInputChange}
            margin="normal"
            fullWidth
            error={error}
          />
          <br />
          <Button
            color="primary"
            variant="contained"
            type="submit"
            onClick={onSubmit}
          >
            Continue
          </Button>
        </DialogContent>
      </Dialog>
    </div>
  );
}
