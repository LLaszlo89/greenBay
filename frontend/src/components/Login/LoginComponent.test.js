import React from "react";
import renderer from "react-test-renderer";
import LoginComponent from "./LoginComponent";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

describe("LoginComponent to redux store ", () => {
  const mockStore = configureStore([]);

  const users = {
    error_message:""
  };

  const store = mockStore({ users });

  it("Should render correctly", () => {
    const tree = renderer.create(
      <Provider store={store}>
        <LoginComponent />
      </Provider>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
