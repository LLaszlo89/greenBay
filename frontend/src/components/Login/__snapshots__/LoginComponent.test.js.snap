// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent to redux store  Should render correctly 1`] = `
<div>
  <h4
    className="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
    style={
      Object {
        "marginTop": "120px",
      }
    }
  >
    Welcome to the coolest web shop on the planet earth!
  </h4>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter"
  >
    To enjoy all its benefits please log in!
  </h6>
  <p
    className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-1"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Login
      </span>
    </button>
  </p>
</div>
`;
